kythe/cxx/indexer/cxx/marked_source.cc

```diff
@@ -697,6 +698,8 @@ void MarkedSourceGenerator::ReplaceMarkedSourceWithTemplateArgumentList(
     }
     *next_arg->mutable_pre_text() = pre_text;
   }
+
+  diags->setClient(PreviousClient, !!OwningPreviousClient.release());
```

Comment:
This should still be done with a scope guard:
```
auto guard = MakeScopeGuard([diags = diags, client = diags->getClient(), owned = diags->takeClient()] {
  diags->setClient(client, owned.release() != nullptr));
});
```

Reply:
I think the lambda expression will capture the smart pointer as const, therefore you cannot call release() inside the body without some proper cast.

Reply:
Fair point, but you should just need to make the lambda mutable:
```
auto guard = MakeScopeGuard([diags = diags, client = diags->getClient(), owned = diags->takeClient()] () mutable {
  diags->setClient(client, owned.release() != nullptr));
});
```

Reply:
I see, I have changed to use scopeguard which looks more elegant